Added New Mode and Function

muratcesmecioglu opened the pull request:
I added a new mod and function for like last x photos of some specified users.

Killerherts commented:
We need testers for this I have seen requests for this feature

LevPasha commented:
Guys wait your tests!

DrCherry commented:
@muratcesmecioglu which mode will test these new features?  I'm running your code in 0.

muratcesmecioglu commented:
@DrCherry i added this feature as mode 6.
https://github.com/LevPasha/instabot.py/pull/355/commits/7e8f46cf56b636a9321aa6e9daa223135265b774

DrCherry commented:
@muratcesmecioglu I've run your code in 0 mode for quite some time with no regression issues.  I then ran it in mode 6 and got this:

`Instabot v1.1.0 started at 20.01.2017 16:31:

Trying to login as doctor_alabaster...

doctor_alabaster login success!
Completed populating user blacklist with IDs
Now username1's photos are liking
Except on get_media!
Traceback (most recent call last):
  File "example.py", line 99, in <module>
    for media in medias:
TypeError: 'int' object is not iterable
Logout: likes - 0, follow - 0, unfollow - 0, comments - 0.
Bot work time: 0:00:30.174991
Logout success!
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/home/scherry/sandbox/instabot.py/src/instabot.py", line 283, in cleanup
    exit(0)
  File "/usr/lib/python3.6/_sitebuiltins.py", line 26, in __call__
    raise SystemExit(code)
SystemExit: 0
`

I'm running python 3.6.0 on Arch Linux.

DrCherry commented on example.py:
```diff
@@ -91,7 +91,7 @@
-    elif mode == 7 :
+    elif mode == 6 :
         last_x_photo = 3;
         userlist = ['username1','username2','username3']
```
I think having the userlist here is problematic.  It should probably be up with the other arrays in the top of the object/class.

muratcesmecioglu replied:
Yes, you are right. There is no checking for user exist or not.
I made this mod for like pictures of my followers who i already know their usernames. So when i fill username array, there aren't be any non existent user.

PeerHoffmann replied:
@muratcesmecioglu In my opinion it would be nice to like pictures from all followers / following / followers and following. It makes sense to use your feature together with the follow function. If you interact more with a person you justed followed the chance for a follow back is much bigger ;-)

davisonio commented:
We need to create a `modes` folder in `src/modes` because there is becoming quite a lot of them.

LevPasha commented:
@davisonio yes, i tired of this modes...

TTAAEELL commented:
This mode is so useful, for me at least. Any progress?

I have issue same as @DrCherry on Python 3.6

muratcesmecioglu commented:
@TTAAEELL Can you try latest codes on Python 2.7?

TTAAEELL commented:
@muratcesmecioglu Works fine on 2.7

Suggestion: Is there any way to add a detection for previously liked photos? 

Example: I run script now and it likes 3 last photos of a user. Then, I run it 12 hours later but that user didn't post new photos, so script skip them and don't try to like again.

TTAAEELL commented:
Another suggestion: Is there any way to scrape users from activity feed. So, bot get a user who liked our photo recently, then like it's last X photos.

LevPasha commented:
@muratcesmecioglu what i should do with this?

muratcesmecioglu commented:
I think now it can work with Python 3.5 and maybe all 3.x versions. I tested with Python 3.5 and 2.7.

@LevPasha Please let allow to other people to do their tests and see if it's work. If they like, you decide to add this mod to your codes. If they don't like or you don't want, this can stay as fork in my github.

@TTAAEELL I don't know there is a detection about liked-or-not status. As far as i know you can see photos you have liked from mobile device but not website. So bot waits 5 minutes in the end. If you have got slow timeline like mine, you can increase that value.

ChipWolf commented:
@muratcesmecioglu could you refactor this based on recent changes to master please

ChipWolf commented:
If this is still an issue, please open a new issue. Thanks!

FabWine commented:
@muratcesmecioglu - is your mode working for like last x photos of some specified users?

muratcesmecioglu commented:
@FabWine Yes. You can find my mode in my repo.

FabWine commented:
Thanks a lot :)  so I put usernames in the example.py?

muratcesmecioglu commented:
Yes. 

- Download instabot from my repo.
- Open example.py with notepad
- Change line54 to   **mode = 6**
- You will find a array in line97. Edit usernames OR remove unnecessery ones. Please be careful for commas and quotes.
**userlist = ['username1','username2','username3']**
- Change username/password placeholders in line15 with your username and password
- Save the file
- Run with python 2.7

FabWine commented:
@muratcesmecioglu Thank you so much :)  I will check it out

FabWine commented:
@muratcesmecioglu getting errors
 File "/Users/TheMinion/Desktop/instabot.py-master 2/example.py", line 7, in <module>
    from instabot import InstaBot
  File "/Users/TheMinion/Desktop/instabot.py-master 2/src/instabot.py", line 4, in <module>
    from userinfo import UserInfo
  File "/Users/TheMinion/Desktop/instabot.py-master 2/src/userinfo.py", line 4, in <module>
    import requests
ImportError: No module named requests

FabWine commented:
@muratcesmecioglu Im getting errrors 

Traceback (most recent call last):
  File "/Users/WineConverterMS/Desktop/instabot.py-master 2/example.py", line 36, in <module>
    '.id','_id','bags'])
  File "/Users/WineConverterMS/Desktop/instabot.py-master 2/src/instabot.py", line 185, in __init__
    self.login()
  File "/Users/WineConverterMS/Desktop/instabot.py-master 2/src/instabot.py", line 225, in login
    self.s.headers.update({'X-CSRFToken': r.cookies['csrftoken']})
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/requests/cookies.py", line 329, in __getitem__
    return self._find_no_duplicates(name)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/requests/cookies.py", line 400, in _find_no_duplicates
    raise KeyError('name=%r, domain=%r, path=%r' % (name, domain, path))
KeyError: "name='csrftoken', domain=None, path=None"

kinggsidd commented:
I’m getting the same error also! Sucks

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: FabWine <notifications@github.com>
Sent: Friday, June 1, 2018 7:46:50 PM
To: instabot-py/instabot.py
Cc: Subscribed
Subject: Re: [instabot-py/instabot.py] Added New Mode and Function (#355)


@muratcesmecioglu<https://github.com/muratcesmecioglu> Im getting errrors

Traceback (most recent call last):
File "/Users/WineConverterMS/Desktop/instabot.py-master 2/example.py", line 36, in
'.id','_id','bags'])
File "/Users/WineConverterMS/Desktop/instabot.py-master 2/src/instabot.py", line 185, in init
self.login()
File "/Users/WineConverterMS/Desktop/instabot.py-master 2/src/instabot.py", line 225, in login
self.s.headers.update({'X-CSRFToken': r.cookies['csrftoken']})
File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/requests/cookies.py", line 329, in getitem
return self._find_no_duplicates(name)
File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/requests/cookies.py", line 400, in _find_no_duplicates
raise KeyError('name=%r, domain=%r, path=%r' % (name, domain, path))
KeyError: "name='csrftoken', domain=None, path=None"

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/instabot-py/instabot.py/pull/355#issuecomment-393975158>, or mute the thread<https://github.com/notifications/unsubscribe-auth/Al9Evu9PmL_IwDMFxj4gmg9awim05m0Oks5t4YwagaJpZM4LjxDR>.

kinggsidd commented:
Hey Fab i just got the same exact error as you!

Did you manage to find out whats up?


ile "example.py", line 6, in <module>

    from src import InstaBot

  File "/Users/SidKing/Desktop/instabot.py-master/src/__init__.py", line 1, in <module>

    from .instabot import InstaBot

  File "/Users/SidKing/Desktop/instabot.py-master/src/instabot.py", line 6, in <module>

    from .userinfo import UserInfo

  File "/Users/SidKing/Desktop/instabot.py-master/src/userinfo.py", line 5, in <module>

    import requests

ImportError: No module named requests



________________________________
From: FabWine <notifications@github.com>
Sent: 01 June 2018 19:35
To: instabot-py/instabot.py
Cc: Subscribed
Subject: Re: [instabot-py/instabot.py] Added New Mode and Function (#355)


@muratcesmecioglu<https://github.com/muratcesmecioglu> getting errors
File "/Users/TheMinion/Desktop/instabot.py-master 2/example.py", line 7, in
from instabot import InstaBot
File "/Users/TheMinion/Desktop/instabot.py-master 2/src/instabot.py", line 4, in
from userinfo import UserInfo
File "/Users/TheMinion/Desktop/instabot.py-master 2/src/userinfo.py", line 4, in
import requests
ImportError: No module named requests

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/instabot-py/instabot.py/pull/355#issuecomment-393972026>, or mute the thread<https://github.com/notifications/unsubscribe-auth/Al9Evqq8VY6MBUH_pE-NKUKnKcbKghC5ks5t4YmHgaJpZM4LjxDR>.

FabWine commented:
@kinggsidd nope

muratcesmecioglu commented:
@FabWine and @kinggsidd 
I updated src/instabot.py file for new instagram. I think instagram changed their code, so bot can't fetch media ids. I updated my function. Can you try again?
And pleae don't forget to install requests module.
